common/src/main/java/org/keycloak/common/util/KeystoreUtil.java

```diff
@@ -105,6 +106,17 @@ public static KeyPair loadKeyPairFromKeystore(String keystoreFile, String storeP
             throw new RuntimeException("Failed to load private key: " + e.getMessage(), e);
         }
     }
+    
+    public static Optional<KeystoreFormat> getKeystoreFormat(String path) {
```

Comment:
It'd be great to have at least some test cases for this method when they are made public.

Reply:
Added a couple of tests.